client/my-sites/plugins/marketplace/marketplace-domain-upsell/index.tsx

```diff
@@ -0,0 +1,198 @@
+import { domainRegistration } from 'calypso/lib/cart-values/cart-items';
+import {
+	MARKETPLACE_FLOW_ID,
+	ANALYTICS_UI_LOCATION_MARKETPLACE_DOMAIN_SELECTION,
+} from 'calypso/my-sites/plugins/marketplace/constants';
+import { getWpComDomainBySiteId } from 'calypso/state/sites/domains/selectors';
+import { fetchSiteDomains } from 'calypso/state/sites/domains/actions';
+import { getSelectedSite } from 'calypso/state/ui/selectors';
+import { getProductsList, isProductsListFetching } from 'calypso/state/products-list/selectors';
+import { fillInSingleCartItemAttributes } from 'calypso/lib/cart-values';
+import Masterbar from 'calypso/layout/masterbar/masterbar';
+import Item from 'calypso/layout/masterbar/item';
+import getPreviousPath from 'calypso/state/selectors/get-previous-path';
+import { HorizontalRule } from '@wordpress/components';
+import ExternalLink from 'calypso/components/external-link';
+import MarketplaceShoppingCart from 'calypso/my-sites/plugins/marketplace/components/marketplace-shopping-cart';
+import theme from 'calypso/my-sites/plugins/marketplace/theme';
+
+/**
+ * Style dependencies
+ */
+import 'calypso/my-sites/plugins/marketplace/marketplace-domain-upsell/style.scss';
+
+const MarketplaceHeaderTitle = styled.h1`
+	font-size: 2rem;
+	margin-bottom: 8px;
+`;
+const MarketplaceHeaderSubTitle = styled.h2`
+	font-size: 0.875rem;
+	margin-bottom: 15px;
+`;
+
+function MarketplaceDomainUpsellHeader() {
+	return (
+		<div>
+			<MarketplaceHeaderTitle className="marketplace-domain-upsell__title wp-brand-font">
+				{ translate( 'Choose a domain' ) }
+			</MarketplaceHeaderTitle>
+			<MarketplaceHeaderSubTitle>
+				{ translate(
+					'Yoast SEO requires a top level domain to index your site and help you rank higher.'
+				) }
+			</MarketplaceHeaderSubTitle>
+			<ExternalLink
+				href="https://yoast.com/domain-names-seo"
+				target="_blank"
+				rel="noopener noreferrer"
+				icon
+			>
+				{ translate( 'Domain names and their influence on SEO' ) }
+			</ExternalLink>
+			<HorizontalRule />
+		</div>
+	);
+}
+
+const getSiteNameFromURL = ( url: string ) => url?.split( '.' )?.[ 0 ] ?? url;
+
+function CalypsoWrappedMarketplaceDomainUpsell(): JSX.Element {
+	const [ selectedDomainProductUUID, setDomainProductUUID ] = useState< string >( '' );
+	const [ selectedDomain, setDomain ] = useState< DomainSuggestions.DomainSuggestion | null >(
+		null
+	);
+
+	const [ isExpandedBasketView, setIsExpandedBasketView ] = useState( false );
+	const { addProductsToCart, replaceProductsInCart, removeProductFromCart } = useShoppingCart();
+	const products = useSelector( getProductsList );
+	const previousPath = useSelector( getPreviousPath );
+	const isFetchingProducts = useSelector( isProductsListFetching );
+	const selectedSite = useSelector( getSelectedSite );
+	const { domain } = useSelector( ( state ) =>
+		getWpComDomainBySiteId( state, selectedSite && selectedSite.ID )
```

Comment:
I wonder if optional chaining could help improve this:

```suggestion
		getWpComDomainBySiteId( state, selectedSite?.ID )
```

Reply:
Thanks I have added this suggestion!